sound/Music.java

```diff
@@ -203,21 +213,51 @@ public void playTune(Tune tune) {
         while (tuneIt.hasNext()) {
             System.out.println("[Line " + (line++) + "] ");
             chord = tuneIt.next();
-            playChord(chord,duration);
+            playChord(chord, duration);
         }
         silence();
     }
-    
+
     /**
      * Stop all tones coming from synthesizer.
      */
     public void silence() {
         Note note;
         int numNotes = noteHistory.size();
-        for (int i=0; i < numNotes; i++) {
+        for (int i = 0; i < numNotes; i++) {
             note = noteHistory.get(i);
-            channels[note.channel()].noteOff(note.pitch(),note.velocityOn());
+            channels[note.channel()].noteOff(note.pitch(), note.velocityOn());
         }
     }
+    /**
+    *@return synth.
+    */
+    public Synthesizer getSynthesizer() {
+       return synth;
+    }
+    /**
+    *@return channels.
+    */
+    public MidiChannel[] getChannels() {
+       return channels;
+    }
+    /**
+     *@return instrument.
+     */
+    public int getInstrument() {
+       return instrument;
+    }
+    /**
+     *@return velocity.
+     */
+    public int getVelocity() {
+       return velocity;
+    }
+    /**
+     *@return numChannels.
+     */
+    public int getNumChannels() {
+       return numChannels;
+    }
```

Comment:
Please explain why these accessor methods are needed?

Reply:
Checkstyle report stated need for these.

Reply:
Checkstyle is an automated tool that assists with identifying potential shortcomings in the code.  Checkstyle only looks at each file in isolation, with no additional context or understanding of the project as a whole.  The messages Checkstyle presents need to be interpreted in context and with a software developer's knowledge and perspective.

(In fact, sometimes the most appropriate way to address a checkstyle error message is to tell checkstyle that the code is written as intended and it should not issue the error.  For example, see: [suppressions.xml](https://github.com/MetroCS/imagelab//blob/master/config/checkstyle/suppressions.xml))

In this particular case, Checkstyle was assuming that instance fields with visibility "package" were actually intended to have visibility "public".  If that wasn't a valid assumption, then the recommendations are suspect.

So we need to look at the project architecture overall to determine what visibility is most appropriate.  In particular, _where are those instance field values used?_  Why was "package" visibility used in the original code?  Do these values need to be accessed or modified from outside the class?  (If not, then there should be no non-private accessors or mutators.)

"Clean code" includes only code that is useful and used.  Adding methods that aren't used makes the code more difficult to understand.

The proposed changes result in changing the access visibility from "package" to "public".

I'm seeking a rationale for that decision versus the change of access visibility from "package" to "private".

Reply:
Looking over the code, it appears that these instance variables are only used within the package sound. They are referred to by other sound objects within a different package(imagelab) but since imagelab does not directly need access to those instances, I believe we can safely remove these getters.

Reply:
Any fields that are used by other objects within the **sound** package still needs accessors, but should have "package" (default) visibility rather than "`public`".

Reply:
I have removed the public modifier and left them as default package access only.

Reply:
> Looking over the code, it appears that these instance variables are only used within the package sound. They are referred to by other sound objects within a different package(imagelab) but since imagelab does not directly need access to those instances, I believe we can safely remove these getters.

I apologize, but I believe I misunderstood this comment as meaning that these instance variables were being used by classes within the _sound_ package other than the Music class.  I just did a quick grep of the code files in the _sound_ package and could only find references to these instance variables inside the Music class itself.  Thus they really should not be visible outside members of the class.  And, since the accessor methods you provided do no processing at all, your original comment--that the methods could be removed--is correct.  These accessor methods should be removed completely.